src/integrator_utils.jl

```diff
@@ -275,3 +275,82 @@ function advance_ode_integrator!(integrator::DDEIntegrator)
     # update prev_idx to index of t and u(t) in solution
     integrator.prev_idx = length(integrator.sol.t)
 end
+
+"""
+    handle_discontinuities!(integrator::DDEIntegrator)
+
+Handle discontinuities at the current time point of `integrator`.
+"""
+function handle_discontinuities!(integrator::DDEIntegrator)
+    # remove all discontinuities at current time point and calculate minimal order
+    # of these discontinuities
+    d = pop!(integrator.opts.d_discontinuities)
+    order = d.order
+    while !isempty(integrator.opts.d_discontinuities) &&
+        top(integrator.opts.d_discontinuities) == integrator.t
+
+        d2 = pop!(integrator.opts.d_discontinuities)
+        order = min(order, d2.order)
+    end
+
+    # remove all discontinuities close to the current time point as well and
+    # calculate minimal order of these discontinuities
+    if typeof(integrator.t) <: AbstractFloat # does not work for units!
+        while !isempty(integrator.opts.d_discontinuities) &&
+            abs(top(integrator.opts.d_discontinuities).t - integrator.t) < 10eps(integrator.t)
+
+            d2 = pop!(integrator.opts.d_discontinuities)
+            order = min(order, d2.order)
+        end
+
+        # also remove all corresponding time stops
+        while !isempty(integrator.opts.tstops) &&
```

Comment:
won't this delete user-chosen timestops?

Reply:
Yes, it will. I'm not sure whether this it necessary, but I thought it might be good to remove discontinuities that are too close to each other, which could happen e.g. with constant lags of `1/3` and `1`. But if we remove close discontinuities we should also remove close time stops, I thought. Of course, this does not differentiate between user-chosen and added time stops (such as newly found discontinuities) - I'm not sure whether this is a problem? We could also remove this completely in the beginning, and just add it if it seems useful for some concrete examples.

Reply:
Oh I see. I guess it doesn't matter. For some reason I thought this could be re-deleted in the future, making it miss these timepoints. But I guess that won't happen.

Reply:
What do you mean by "re-deleted in the future"? The integrator won't stop at these future timepoints, but I thought this might be OK since they are so close.

Reply:
Yeah, that's fine.